src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -102,62 +80,23 @@ public bool CreateNoWindow
 
                     // Manual use of IDictionaryEnumerator instead of foreach to avoid DictionaryEntry box allocations.
                     IDictionaryEnumerator e = envVars.GetEnumerator();
-                    try
+                    while (e.MoveNext())
                     {
-                        while (e.MoveNext())
-                        {
-                            DictionaryEntry entry = e.Entry;
-                            _environmentVariables.Add((string)entry.Key, (string)entry.Value);
-                        }
-                    }
-                    finally
-                    {
-                        (e as IDisposable)?.Dispose();
```

Comment:
It comes back to me now. Well, in theory we could replace this with a Dictionary at some point as it's not in the public surface area (subject to any threading guarantees)

Although something else comes back to me -- someone did change it to a dictionary in ?3.5 and it broke folks -- including MSBuild who was unwisely casting to a Hashtable. Clearly, it's changed back so I guess changing now isn't going to happen.